toolkit/tools/internal/azureblobstorage/azureblobstorage.go

```diff
@@ -22,6 +25,47 @@ const (
 	AzureCLIAccess         = 2
 )
 
+// ErrDownloadInvalidResponse404 is returned when the download response is 404.
+// Example for blob (similar response and code for container case):
+// RESPONSE 404: 404 The specified blob does not exist.
+// ERROR CODE: BlobNotFound
+var ErrDownloadInvalidResponse404 = errors.New("RESPONSE 404")
+
+// ParseAzureBlobStorageURL parses an Azure Blob Storage URL and extracts storage account, container, and optionally blob information.
+func ParseAzureBlobStorageURL(urlStr string) (storageAccountName, containerName, blobName string, err error) {
+	parsedURL, err := url.Parse(urlStr)
+	if err != nil {
+		return "", "", "", fmt.Errorf("failed to parse URL: %w", err)
+	}
+
+	if !strings.HasSuffix(parsedURL.Host, ".blob.core.windows.net") {
+		return "", "", "", fmt.Errorf("not a common Azure Blob Storage URL format " +
+			"(expected <storage_account>.blob.core.windows.net)")
+	}
+
+	// Extract storage account from hostname (e.g., "mystorageaccount.blob.core.windows.net")
+	hostParts := strings.Split(parsedURL.Host, ".")
+	if len(hostParts) < 4 {
```

Comment:
Shouldn't there be at least 5 parts after the split:
- <storage name>,
- "blob",
- "core",
- "windows",
- "net".

Reply:
Changed via: https://github.com/microsoft/azurelinux/pull/14612